scripts/thumbnail-generator/generate_thumbnail.py

```diff
@@ -0,0 +1,302 @@
+    clippingPlanes = Gf.Vec2f(nearClip, farClip)
+    camera_prim.GetClippingRangeAttr().Set(clippingPlanes)
+
+    if args.verbose:
+        print("Calculating clipping planes... " + str(clippingPlanes))
+
+    return cameraZ
+
+def get_bounding_box(subject_stage):
+    bboxCache = UsdGeom.BBoxCache(Usd.TimeCode.Default(), [UsdGeom.Tokens.default_])
+    # Compute the bounding box for all geometry under the root
+    root = subject_stage.GetPseudoRoot()
+    return bboxCache.ComputeWorldBound(root).GetBox()
+
+def get_distance_to_camera(min_bound, max_bound, camera_prim):
+    focal_length = camera_prim.GetFocalLengthAttr().Get()
+    horizontal_aperture = camera_prim.GetHorizontalApertureAttr().Get()
+    vertical_aperture = camera_prim.GetVerticalApertureAttr().Get()
+
+    distance_to_capture_horizontal = calculate_field_of_view_distance(horizontal_aperture, (max_bound[0] - min_bound[0]) * 10, focal_length)
+    distance_to_capture_vertical = calculate_field_of_view_distance(vertical_aperture, (max_bound[1] - min_bound[1]) * 10, focal_length)
+
+    return max(distance_to_capture_horizontal, distance_to_capture_vertical)
+
+def calculate_field_of_view_distance(sensor_size, object_size, focal_length):
+    return calculate_camera_distance(object_size, calculate_field_of_view(focal_length, sensor_size))
+    
+def calculate_field_of_view(focal_length, sensor_size):
+    # Focal length and sensor size should be in the same units (e.g., mm)
+    field_of_view = 2 * math.atan(sensor_size / (2 * focal_length))
+    return field_of_view
+
+def calculate_camera_distance(subject_size, field_of_view):
+    # Subject size and field of view should be in the same units (e.g., mm and degrees)
+    distance = (subject_size / 2) / math.tan(field_of_view / 2)
+    return distance
+
+def get_camera_z_translation(distance):
+    return Gf.Vec3d(0, 0, distance)
+
+def apply_camera_translation(camera_stage, camera_prim, camera_translation):
+    xformRoot = UsdGeom.Xformable(camera_prim.GetPrim())
+    translateOp = None
+    # Go through each operation in the xformable schema
+    for op in xformRoot.GetOrderedXformOps():
+        # If the operation is a translate operation, we've found our operation
+        if op.GetOpType() == UsdGeom.XformOp.TypeTranslate:
+            translateOp = op
+            break
+
+    # If no translate operation exists, create one
+    if translateOp is None:
+        translateOp = xformRoot.AddTranslateOp(UsdGeom.XformOp.PrecisionDouble)
+
+    translateOp.Set(camera_translation)
+    camera_stage.Save()
+
+def sublayer_subject(camera_stage, input_file):
+    camera_stage.GetRootLayer().subLayerPaths = [input_file]
+    camera_stage.GetRootLayer().Save()
+
+def take_snapshot(image_name):
+    renderer = get_renderer()
+    cmd = ['usdrecord', '--camera', 'MainCamera', '--imageWidth', str(args.width), '--renderer', renderer, 'camera.usda', image_name]
+    run_os_specific_usdrecord(cmd)
+    os.remove("camera.usda")
```

Comment:
Maybe call this "default_thumnail.usda" or something? And use a constant for it defined at the top of the file rather than hard coding it in several places. As mentioned in another comment, I don't think y_up.usda should exist at all.

Reply:
makes sense, will update